demos/common/cpp/models/src/hpe_model_associative_embedding.cpp

```diff
@@ -0,0 +1,207 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include "models/associative_embedding_decoder.h"
+#include "models/hpe_model_associative_embedding.h"
+
+#include <utils/common.hpp>
+#include <utils/ocv_common.hpp>
+#include <ngraph/ngraph.hpp>
+
+using namespace InferenceEngine;
+
+const cv::Vec3f HpeAssociativeEmbedding::meanPixel = cv::Vec3f::all(128);
+const float HpeAssociativeEmbedding::detectionThreshold = 0.1f;
+const float HpeAssociativeEmbedding::tagThreshold = 1.0f;
+const float HpeAssociativeEmbedding::delta = 0.0f;
+const bool HpeAssociativeEmbedding::useDetectionVal = true;
+const bool HpeAssociativeEmbedding::doAdjust = true;
+const bool HpeAssociativeEmbedding::doRefine = true;
+const bool HpeAssociativeEmbedding::ignoreTooMuch = true;
+
+HpeAssociativeEmbedding::HpeAssociativeEmbedding(const std::string& modelFileName, double aspectRatio,
+    int targetSize, float confidenceThreshold) :
+    ModelBase(modelFileName),
+    aspectRatio(aspectRatio),
+    targetSize(targetSize),
+    confidenceThreshold(confidenceThreshold) {
+}
+
+void HpeAssociativeEmbedding::prepareInputsOutputs(CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    if (inputShapes.size() != 1)
+        throw std::runtime_error("Demo supports topologies only with 1 input");
+    inputsNames.push_back(inputShapes.begin()->first);
+    SizeVector& inSizeVector = inputShapes.begin()->second;
+    if (inSizeVector.size() != 4 || inSizeVector[0] != 1 || inSizeVector[1] != 3)
+        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
+    InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
+    inputInfo.setPrecision(Precision::U8);
+    inputInfo.getInputData()->setLayout(Layout::NCHW);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    const OutputsDataMap& outputInfo = cnnNetwork.getOutputsInfo();
+    if (outputInfo.size() != 3)
+        throw std::runtime_error("Demo supports topologies only with 3 outputs");
+    for (const auto& outputLayer: outputInfo) {
+        outputLayer.second->setPrecision(Precision::FP32);
+        outputsNames.push_back(outputLayer.first);
```

Comment:
Relaying on order of `outputInfo` traversing is equivalent to hard coding names of outputs, but the second option is clearer. Please implement it.

Reply:
Added `findLayerByName` function.